File: src/AppRegistryService.Contract/Responses/AppReleaseInfo.cs
using AppRegistryService.Contract.Models;

namespace AppRegistryService.Contract.Responses;

/// <summary>
/// Defines an application release information.
/// </summary>
public sealed class AppReleaseInfo
{
    /// <summary>
    /// Unique release identifier.
    /// </summary>
    public Guid Id { get; set; }

    /// <summary>
    /// Application identifier link.
    /// </summary>
    public Guid AppId { get; set; }

    /// <summary>
    /// Release version.
    /// </summary>
    public Version? Version { get; set; }

    /// <summary>
    /// Mimumim supported operating system version.
    /// </summary>
    public Version? MinimumOSVersion { get; set; }

    /// <summary>
    /// Release publish date.
    /// </summary>
    public DateOnly? PublishDate { get; set; }

    /// <summary>
    /// Release notes.
    /// </summary>
    public string? Notes { get; set; }

    /// <summary>
    /// Release level.
    /// </summary>
    public ReleaseLevel Level { get; set; }
}

File: web/src/responses/AppReleaseInfo.ts
import ReleaseLevel from '../models/ReleaseLevel';

/**
 * Defines an application release information.
 */
export interface AppReleaseInfo {
    /**
     * Unique release identifier.
     */
    id: string;

    /**
     * Application identifier link.
     */
    appId: string;

    /**
     * Release version.
     */
    version?: string;

    /**
     * Minimum supported operating system version.
     */
    minimumOSVersion?: string;

    /**
     * Release publish date.
     */
    publishDate?: Date;

    /**
     * Release notes.
     */
    notes?: string;

    /**
     * Release level.
     */
    level: ReleaseLevel;

    /**
     * Is this release mandatory.
     */
    isMandatory: boolean;
}